src/azure-cli-testsdk/azure/cli/testsdk/checkers.py

```diff
@@ -59,6 +60,20 @@ def __call__(self, execution_result):
                                                   execution_result.output)
 
 
+class JMESPathPatternCheck(object):  # pylint: disable=too-few-public-methods
+        def __init__(self, query, expected_result):
+            self._query = query
+            self._expected_result = expected_result
+
+        def __call__(self, execution_result):
+            json_value = execution_result.get_output_in_json()
+            actual_result = jmespath.search(self._query, json_value,
+                                            jmespath.Options(collections.OrderedDict))
+            if not re.match(self._expected_result, str(actual_result), re.IGNORECASE):
+                raise JMESPathCheckAssertionError(self._query, 'aaa', actual_result,
+                                                  execution_result.output)
```

Comment:
@troydai I added this back in. Looks like it was missed from the prev. vcr_test_base